### ISSUE TITLE
Undefined injected services even after #111 or #122

### ISSUE THREAD

**Issue by dlecan** [ocombe/ocLazyLoad#124]
I'm trying to include ocLazyLoad module in my current Angular application, but I can't succeed.

Context:
- Angular 1.2.28
- Many dependencies: ui-router, angular-translate, angular-google-maps, ui-bootstrap, ...
- Many internal modules
- Many Karma + Jasmine tests on PhantomJS

#### Add oclazyload.js script to the build
All tests are OK

#### Add oc.LazyLoad module dependency to my app
Some tests fail (~1/3) because of undefined injected services.
These can be internal services of Angular ones ($httpBacked)
Undefined services are consistant between tests executions.

As the errors encountered look like thoses described in #111 or #122, I tried with ocLazyLoad from master (98aad14141e2eae1d04f9fc1fe09d85cd4b14713).

#### Add oclazyload.js script to the build
Even if oc.LazyLoad module is not loaded, all my tests don't run at all! No error log, nothing to understand what happens.

```
INFO [karma]: Karma v0.12.24-1 server started at http://localhost:49221/
INFO [launcher]: Starting browser PhantomJS
INFO [PhantomJS 1.9.8 (Linux)]: Connected on socket xxWbU2HShENLlRvi9jck with id 43520942

PhantomJS 1.9.8 (Linux): Executed 0 of 60 ERROR (0.056 secs / 0 secs)
```

What's wrong with my project ?

**Comment by ocombe** [comment]
Hi !
ocLazyLoad uses promises and they are only resolved when Angular triggers a digest.
If you take a look at my tests you'll see that I use a setInterval to trigger digests once every x ms to make sure that all promises get resolved, otherwise the tests will never end (that's probably what happened to you here).
If you find a better way to do it than to use setInterval, I'll take it :)

**Comment by dlecan** [comment]
I understand you wrote some hacks to make your tests run. But I don't understand your answer according to my context and my Angular application.

Do you mean I should improve my unit tests to support oclazyload script or module when they are included in **my** application ?

**Comment by dlecan** [comment]
@ocombe some clue to help me to diagnose what happens? Is there a debug mode I can enable which may help to understand what is wrong in my application?
Thank you

**Comment by ocombe** [comment]
Yes, $ocLazyLoad returns a promise that can be rejected if there is an error. If you set debug to true, $ocLazyLoad will also log all errors & list all events to the console.

```js
$ocLazyLoadProvider.config({
    debug: true
});
```

And regarding the tests, if you want to make your tests run with ocLazyLoad in your app, You'll have to find a way to execute the $digest multiple times to execute the promises.
You could also stub $q to make it auto resolve, it might work

**Comment by ocombe** [comment]
Closing this due to lack of activity, ask me if you need it reopened.
